tree/ntuple/v7/inc/ROOT/RDaos.hxx

```diff
@@ -40,6 +40,46 @@ namespace ROOT {
 namespace Experimental {
 namespace Detail {
 
+std::string inline GetDaosError(int err)
+{
+   return d_errstr(err);
+}
+
+struct RDaosObjectId {
+   using ObjectIndex_t = std::uint64_t;
+   daos_obj_id_t fData;
+   RDaosObjectId(std::uint64_t ntupleIndex, ObjectIndex_t objectIndex)
+      : fData{static_cast<decltype(daos_obj_id_t::lo)>(objectIndex),
+              static_cast<decltype(daos_obj_id_t::hi)>(ntupleIndex)}
+   {
+   }
+   [[nodiscard]] daos_obj_id_t Get() const { return fData; }
+};
+
+struct RDaosKey {
+   using DistributionKey_t = std::uint64_t;
+   using AttributeKey_t = std::uint64_t;
+   RDaosObjectId fOid;
+   DistributionKey_t fDkey;
+   AttributeKey_t fAkey;
+};
+
+struct RDaosIov {
+   d_iov_t fIov{};
+
+   RDaosIov(void *buffer, size_t size) : fIov{buffer, size, size} {}
+   void Set(void *buffer, size_t size) { d_iov_set(&fIov, buffer, size); }
+   [[nodiscard]] d_iov_t Get() const { return fIov; }
+   [[nodiscard]] void *GetBuffer() const { return fIov.iov_buf; }
+   [[nodiscard]] size_t GetBufferSize() const { return fIov.iov_buf_len; }
+};
```

Comment:
More in general, and if it doesn't hurt performance, maybe we could the standard library type `std::span<unsigned char *>` instead of providing this type.